src/vs/workbench/electron-sandbox/window.ts

```diff
@@ -266,6 +269,17 @@ export class NativeWindow extends Disposable {
 		)(e => this.onDidChangeMaximized(e)));
 
 		this.onDidChangeMaximized(this.environmentService.configuration.maximized ?? false);
+
+		// Detect panel position to determine minimum width
+		this._register(this.layoutService.onPanelPositionChange(pos => {
+			this.onDidPanelPositionChange(positionFromString(pos));
+		}));
+		this.onDidPanelPositionChange(this.layoutService.getPanelPosition());
+
+		this._register(this.layoutService.onPartVisibilityChange(() => {
+			this.onDidPanelVisibilityChange(!this.layoutService.isPanelHidden());
+		}));
+		this.onDidPanelVisibilityChange(!this.layoutService.isPanelHidden());
```

Comment:
I am not a big fan of sending an IPC message to the main side right on startup to tell the correct minimal window size to be honest. Can we not read this setting from the main side and configure the minimal width properly already? Where is this state stored, is it a setting? We can read global user settings from the main side, but not UI state or workspace settings.

Reply:
I don't think the default minsize is stored in any setting right now. Each part has a constant minsize within its own ts file, and the window also has its own constant minsizes right now that doesn't respect the parts' minsizes. So for now, all of these minsizes are constants and hardcoded into the program. I'm not sure whether these constants, which all represent the layout, should be stored somewhere directly accessible by the main program.